plugins/obs-webrtc/whip-output.cpp

```diff
@@ -266,6 +271,47 @@ bool WHIPOutput::Setup()
 	return true;
 }
 
+void WHIPOutput::ParseLinkHeader(std::string val,
+				 std::vector<rtc::IceServer> &iceServers)
+{
+	size_t pos = 0;
+	std::string url, username, password;
+
+	auto extractValue = [](std::string input) -> std::string {
+		auto head = input.find("\"") + 1;
+		auto tail = input.find_last_of("\"");
+
+		if (head == std::string::npos || tail == std::string::npos) {
+			return "";
+		}
+		return input.substr(head, tail - head);
+	};
+
+	while ((pos = val.find(";")) != std::string::npos) {
+		auto token = val.substr(0, pos);
+		if (token.find("turn:", 0) == 0) {
+			url = token;
```

Comment:
Thank you for adding TURN support to WHIP!
I have confirmed that the implementation is working successfully with our SFU after making some minor modifications (please refer to the feedback below).

---

Seeing https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link, the handling of `<` and `>` seems necessary as the doc says "The URI (absolute or relative) must be enclosed between `<` and `>`".

Besides, the doc also says "You can specify multiple links separated by commas". So it would be great if you could add comma handling to support multiple links.
(I know that the current implementation can handle multiple links if SFU returns multiple, duplicate LINK HTTP header fields. However, some HTTP server libraries (e.g., [cowboy](https://github.com/ninenines/cowboy) which is a famous library in Erlang) does not allow response headers having duplicate fields.)

Reply:
Thanks for testing @sile!

I will fix that tomorrow and ping you for another test :)

Reply:
Did you already make modifications to the C++ code or your TURN service? If you modified OBS would you be able to push to my fork?

I gave you write permission to my fork.

Reply:
I modified our SFU. However, if it's okay, I'm willing to fix the OBS code and push the patch to your fork 😃

Reply:
Pushed: https://github.com/obsproject/obs-studio/pull/10524/commits/fd771428aa538c2ddc37860e17963e2a21ba8d59

Feel free to modify (or discard) the code as I am not familiar with C++ and OBS 😅

Reply:
@sile Your code is fantastic! I am really impressed how quickly you were able to do that not being familiar with C++ and OBS.

I squashed the commits down into one and fixed some linter errors. Beyond that everything looks perfect to me.